src/Microsoft.DotNet.TestFramework/Microsoft.DotNet.TestFramework.TestScenario.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Cli.Utils;
+
+namespace Microsoft.DotNet.TestFramework
+{
+    public class TestScenario
+    {
+        private static IDictionary<string, TestScenario> _scenarioCache = new Dictionary<string, TestScenario>();
+
+        private TestScenario(string testSourceRoot, bool skipRestore, bool skipBuild)
+        {
+            SourceRoot = testSourceRoot;
+            Projects = GetAllProjects(SourceRoot);
+
+            if (!skipRestore)
+            {
+                Restore();
+            }
+
+            if (!skipBuild)
+            {
+                Build();
+            }
+        }
+
+        private IEnumerable<string> Projects
+        {
+            get; set;
+        }
+
+        public string SourceRoot
+        {
+            get;
+            private set;
+        }
+
+        public static TestScenario Create(string testSourceRoot, bool skipRestore = false, bool skipBuild = false)
```

Comment:
These projects should be pre-built and pre-restored. Shouldn't the defaults reflect that?